http-client/src/main/java/io/airlift/http/client/spnego/SpnegoAuthentication.java

```diff
@@ -147,6 +132,15 @@ public URI getURI()
             public void apply(Request request)
             {
                 String servicePrincipal = makeServicePrincipal(remoteServiceName, normalizedUri.getHost());
+                LoginContext loginContext = getLoginContext();
+                Subject subject = loginContext.getSubject();
+                Principal clientPrincipal = subject.getPrincipals().iterator().next();
```

Comment:
This can throw if there are no principals. It should probably have an explicit check to throw a proper error message